### DISCUSSION TITLE
Overlap x-axis data

### DISCUSSION THREAD

**Discussion by shamilh2o** [h2oai/wave#1529] [Q&A] [answered]
Hey, Is it capable to align x axis data to prevent overlap which shown below? 
I am trying with wave SDK 0.21.1. I used wave tour application to reproduce this error.

Thanks!

![image](https://user-images.githubusercontent.com/101506482/177505323-c7e7530d-346d-46e1-8333-f728babe56ce.png)
![image](https://user-images.githubusercontent.com/101506482/177537285-abb7df76-8d07-4e3e-ac02-af3a23a06fbe.png)

**Comment by Mathanraj-Sharma** [comment, marked as answer]
@shamilh2o I can suggest two workarounds

- try adjusting the box width `box="1 1 8 5"`. 
![image](https://user-images.githubusercontent.com/30664910/177544357-8961c14c-93ac-4bff-9943-78aee05a6e8e.png)


- or line break the field names
```python
q.page.add('example', ui.plot_card(
        box='1 1 4 5',
        title='Intervals, groups',
        data=data('day type time', 10, rows=[
            ('aaaaaaaaaaa\naaaaaaaaaaaaaa','series1', 2800),
            ('aaaaaaaaaaa\naaaaaaaaaaaaaa','series2', 2260),
            ('bbbbbbbbbbb\nbbbbbbbbbbbbbb','series1', 1800),
            ('bbbbbbbbbbb\nbbbbbbbbbbbbbb','series2', 1300),
            ('ccccccccccc\ncccccccccccccc','series1', 950),
            ('ccccccccccc\ncccccccccccccc','series2', 900),
            ('ddddddddddd\ndddddddddddddd','series1', 500),
            ('ddddddddddd\ndddddddddddddd','series2', 390),
            ('eeeeeeeeeee\neeeeeeeeeeeeee','series1', 170),
            ('eeeeeeeeeee\neeeeeeeeeeeeee','series2', 100),
        ]),
        plot=ui.plot([
                ui.mark(
                    type='interval', 
                    x='=day', 
                    y='=time', 
                    color='=type', 
                    dodge='auto', 
                    y_min=0,
                )
            ])
    ))
```

![image](https://user-images.githubusercontent.com/30664910/177544126-b6f6b68c-1b35-44d2-a3de-7b02f92be431.png)

**Reply by shamilh2o** [reply]
Thanks for the suggestion @Mathanraj-Sharma.

The actual use case I needed this feature is to display a plot with values for particular dates as shown here. 
![image](https://user-images.githubusercontent.com/101506482/177546927-778c0cca-603d-40f8-88f6-0bc0361c4d94.png)

Will try with your suggestions. 
Thank you,

**Reply by Mathanraj-Sharma** [reply]
@mturoci @lo5 

Is there any way to tilt the x-axis label? Or to wrap them when they are overlapping?

**Reply by Mathanraj-Sharma** [reply]
@sandruparo @shihan007  FYI

**Comment by mturoci** [comment]
@Mathanraj-Sharma's suggestions are good.

The root problem is that you want to show too much data, making the plot unreadable. Instead, I would suggest grouping the data by e.g. year or month. So instead of redundant 2022-06-day, you would have just day in case of the group by month etc. This would reduce the axis space and give cleaner results.

**Reply by shamilh2o** [reply]
Thank you @mturoci. Yes, could able to overcome this by aggregating data in weekly manner. By the way in common manner @Mathanraj-Sharma's suggestion too useful to overcome this issue. 
Thank you both!
